include/natalie/value.hpp

```diff
@@ -80,7 +86,21 @@ class Value {
     }
 
     bool operator!=(Value other) const {
-        return !(*this == other);
+        if (is_fast_integer()) {
+            if (other.is_fast_integer())
+                return get_fast_integer() != other.get_fast_integer();
+            if (other.is_fast_float())
+                return get_fast_integer() != other.get_fast_float();
+            return true;
+        }
+        if (is_fast_float()) {
+            if (other.is_fast_integer())
+                return get_fast_integer() != other.get_fast_integer();
+            if (other.is_fast_float())
+                return get_fast_float() != other.get_fast_float();
+            return true;
+        }
+        return m_object != other.object();
```

Comment:
I don't think this duplication is neccesary. This *should* be equivalent to boolean NOT of `==`.